[[ Community Docs ]] added defaultFolder help

matthewmaier opened the pull request:
- I got hung up for hours because I didn't think of "go" using the defaultFolder
- now there's a little code snippet that will ensure the defaultFolder is set to the stack's folder
- it also can be easily modified in other ways

matthewmaier committed:
[[ Community Docs ]] added defaultFolder help

- I got hung up for hours because I didn't think of "go" using the defaultFolder
- now there's a little code snippet that will ensure the defaultFolder is set to the stack's folder
- it also can be easily modified in other ways

livecodeali commented on docs/dictionary/command/go.lcdoc:
```diff
@@ -36,6 +36,13 @@ go back 7
+set the itemDelimiter to "/"
+put item 1 to -2 of tStackFile into tStackFolder
+set the defaultFolder to tStackFolder
+go stack somethingElse.livecode
```
Could you make sure this compiles in strict mode please? I.e. declare the variables and put `somethingElse.livecode` in quotes? Also I think a line of comment like 

```
-- somethingElse.livecode is a stackfile in the same folder as this stack
```

would help explain what the example is for.

asayd replied:
It might be a good idea to add `the effective` to the first statement. That would cover cases when you are working in a substack:

```
put the effective filename of me into tStackFile
```

livecodeali commented on docs/dictionary/command/go.lcdoc:
```diff
@@ -82,6 +89,10 @@ go stack "My Stack"
 
+If you don't include a file path the <go> command will look in
+the defaultFolder. If you're in the IDE (not a standalone) the
+defaultFolder will probably start off as a system folder.
```
I think this should be rephrased slightly to 

```
 Otherwise, you must specify the stack's file path.

If the specified file path is relative, the <go> command will look in the <defaultFolder>.
```

It is the responsibility of the entry for the `defaultFolder` to describe what it starts off as in various modes (indeed it does do this), so I would remove the last sentence here. You'll need to add `defaultFolder` to the list of references.

matthewmaier committed:
[[ Community Docs ]] made suggested changes

- quotes around file name
- rephrased narrative
- added defaultFolder to references
- added [effective] before filename

livecodeali commented on docs/dictionary/command/go.lcdoc:
```diff
@@ -36,6 +36,14 @@ go back 7
 Example:
 go invisible stack "Preferences"
 
+Example:
```
Could you also declare the variables `local tStackFile, tStackFolder` here please?

matthewmaier committed:
declared local variables

livecodeali commented:
@matthewmaier Thanks!
@livecode-vulcan review ok b41f1c4

livecode-vulcan commented:
💙 review by @livecodeali ok b41f1c48e73f34239634596e7ca3dbabd428715a

livecode-vulcan commented:
😎 test success b41f1c48e73f34239634596e7ca3dbabd428715a
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/131)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/131)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/131)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/131)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/131)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/131)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/131)
